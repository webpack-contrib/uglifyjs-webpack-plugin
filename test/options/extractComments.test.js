import webpack from '../helpers/compiler';
import UglifyJsPlugin from '../../src/index';

describe('Options', () => {
  describe('extract', () => {
    test('{String}', async () => {
      const config = {
        entry: {
          1: './extract/1.js',
          main: './entry.js',
        },
        plugins: [
          new UglifyJsPlugin({
            extractComments: 'all',
          }),
        ],
      };

      const stats = await webpack('extract/entry2.js', config);
      const { assets } = stats.compilation;

      expect(assets['1.bundle.js'].source()).toMatchSnapshot();
      expect(assets['1.bundle.js.LICENSE'].source()).toMatchSnapshot();

      expect(assets['main.bundle.js'].source()).toMatchSnapshot();
      expect(assets['main.bundle.js.LICENSE'].source()).toMatchSnapshot();
    });

    test('{Regex}', async () => {
      const config = {
        entry: {
          1: './extract/1.js',
          main: './entry.js',
        },
        plugins: [
          new UglifyJsPlugin({
            extractComments: /.*/,
          }),
        ],
      };

      const stats = await webpack('extract/entry2.js', config);
      const { assets } = stats.compilation;

      expect(assets['1.bundle.js'].source()).toMatchSnapshot();
      expect(assets['1.bundle.js.LICENSE'].source()).toMatchSnapshot();

      expect(assets['main.bundle.js'].source()).toMatchSnapshot();
      expect(assets['main.bundle.js.LICENSE'].source()).toMatchSnapshot();
    });

    test('{Object}', async () => {
      const config = {
        entry: {
          1: './extract/1.js',
          2: './extract/2.js',
          3: './extract/3.js',
          4: './extract/4.js',
          main: './extract/entry.js',
        },
        plugins: [
          new UglifyJsPlugin({
            exclude: /runtime/,
            uglifyOptions: {
              output: {
                comments: 'all',
              },
            },
            extractComments: {
              condition: /.*/,
              filename: 'extracted.js',
            },
          }),
        ],
      };

      const stats = await webpack('extract/entry.js', config);
      const { assets } = stats.compilation;

      expect(assets['1.bundle.js'].source()).toMatchSnapshot('1.bundle.js');
      expect(assets['2.bundle.js'].source()).toMatchSnapshot('2.bundle.js');
      expect(assets['3.bundle.js'].source()).toMatchSnapshot('3.bundle.js');
      expect(assets['4.bundle.js'].source()).toMatchSnapshot('4.bundle.js');
      expect(assets['main.bundle.js'].source()).toMatchSnapshot(
        'main.bundle.js'
      );

      expect(assets['extracted.js'].source()).toMatchSnapshot('extracted.js');
    });

    test('{Object<{Function}>}', async () => {
      const config = {
        plugins: [
          new UglifyJsPlugin({
            extractComments: {
              condition: true,
              filename(file) {
                return file.replace(/(\.\w+)$/, '.license$1');
              },
              banner(file) {
                return `License information can be found in ${file}`;
              },
            },
          }),
        ],
      };

      const stats = await webpack('extract/entry2.js', config);
      const { assets } = stats.compilation;

      expect(assets['main.bundle.js'].source()).toMatchSnapshot();
      expect(assets['main.bundle.license.js'].source()).toMatchSnapshot();
    });

    test('{Function}', async () => {
      const config = {
        entry: {
          1: './extract/1.js',
          main: './entry.js',
        },
        plugins: [
          new UglifyJsPlugin({
            extractComments: () => true,
          }),
        ],
      };

      const stats = await webpack('extract/entry2.js', config);
      const { assets } = stats.compilation;

      expect(assets['1.bundle.js'].source()).toMatchSnapshot();
      expect(assets['1.bundle.js.LICENSE'].source()).toMatchSnapshot();
      expect(assets['main.bundle.js'].source()).toMatchSnapshot();
      expect(assets['main.bundle.js.LICENSE'].source()).toMatchSnapshot();
    });
  });
});
