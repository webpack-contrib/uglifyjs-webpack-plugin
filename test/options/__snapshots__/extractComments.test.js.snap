// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1.bundle.js 1`] = `
"/*! For license information please see extracted.js */
(window.webpackJsonp=window.webpackJsonp||[]).push([[0],[
/* 0 */
/***/function(n,w){}],[[0,5,0]]]);
//# sourceMappingURL=1.bundle.js.map"
`;

exports[`2.bundle.js 1`] = `
"/*! For license information please see extracted.js */
(window.webpackJsonp=window.webpackJsonp||[]).push([[1],[
/* 0 */
/* 1 */
/***/,
/* 1 */
/***/function(n,w){}],[[1,5,1]]]);
//# sourceMappingURL=2.bundle.js.map"
`;

exports[`3.bundle.js 1`] = `
"/*! For license information please see extracted.js */
(window.webpackJsonp=window.webpackJsonp||[]).push([[2],{
/***/2:
/***/function(n,w){}},[[2,5,2]]]);
//# sourceMappingURL=3.bundle.js.map"
`;

exports[`4.bundle.js 1`] = `
"/*! For license information please see extracted.js */
(window.webpackJsonp=window.webpackJsonp||[]).push([[3],{
/***/3:
/***/function(n,w){}},[[3,5,3]]]);
//# sourceMappingURL=4.bundle.js.map"
`;

exports[`Options extract {Function} 1`] = `
"/*! For license information please see 1.bundle.js.LICENSE */
(window.webpackJsonp=window.webpackJsonp||[]).push([[0],[function(n,w){}],[[0,2,0]]]);
//# sourceMappingURL=1.bundle.js.map"
`;

exports[`Options extract {Function} 2`] = `
"/* 0 */

/***/
"
`;

exports[`Options extract {Function} 3`] = `
"/*! For license information please see main.bundle.js.LICENSE */
(window.webpackJsonp=window.webpackJsonp||[]).push([[1],[,function(o,n){o.exports=function(){console.log(7)}}],[[1,2,1]]]);
//# sourceMappingURL=main.bundle.js.map"
`;

exports[`Options extract {Function} 4`] = `
"/* 0 */

/* 1 */

/***/

/* 1 */

/***/
"
`;

exports[`Options extract {Object<{Function}>} 1`] = `
"/*! License information can be found in main.bundle.license.js */
(window.webpackJsonp=window.webpackJsonp||[]).push([[0],[function(n,w){}],[[0,1,0]]]);
//# sourceMappingURL=main.bundle.js.map"
`;

exports[`Options extract {Object<{Function}>} 2`] = `
"/* 0 */

/***/

/** @preserve This is comment 2 */

/***/
"
`;

exports[`Options extract {Regex} 1`] = `
"/*! For license information please see 1.bundle.js.LICENSE */
(window.webpackJsonp=window.webpackJsonp||[]).push([[0],[function(n,w){}],[[0,2,0]]]);
//# sourceMappingURL=1.bundle.js.map"
`;

exports[`Options extract {Regex} 2`] = `
"/* 0 */

/***/
"
`;

exports[`Options extract {Regex} 3`] = `
"/*! For license information please see main.bundle.js.LICENSE */
(window.webpackJsonp=window.webpackJsonp||[]).push([[1],[,function(o,n){o.exports=function(){console.log(7)}}],[[1,2,1]]]);
//# sourceMappingURL=main.bundle.js.map"
`;

exports[`Options extract {Regex} 4`] = `
"/* 0 */

/* 1 */

/***/

/* 1 */

/***/
"
`;

exports[`Options extract {String} 1`] = `
"/*! For license information please see 1.bundle.js.LICENSE */
(window.webpackJsonp=window.webpackJsonp||[]).push([[0],[function(n,w){}],[[0,2,0]]]);
//# sourceMappingURL=1.bundle.js.map"
`;

exports[`Options extract {String} 2`] = `
"/* 0 */

/***/
"
`;

exports[`Options extract {String} 3`] = `
"/*! For license information please see main.bundle.js.LICENSE */
(window.webpackJsonp=window.webpackJsonp||[]).push([[1],[,function(o,n){o.exports=function(){console.log(7)}}],[[1,2,1]]]);
//# sourceMappingURL=main.bundle.js.map"
`;

exports[`Options extract {String} 4`] = `
"/* 0 */

/* 1 */

/***/

/* 1 */

/***/
"
`;

exports[`extracted.js 1`] = `
"/* 0 */

/***/

/* 0 */

/* 1 */

/***/

/* 1 */

/***/

/***/

/***/

/***/

/***/

/***/

/***/
"
`;

exports[`main.bundle.js 1`] = `
"/*! For license information please see extracted.js */
(window.webpackJsonp=window.webpackJsonp||[]).push([[4],{
/***/4:
/***/function(n,w){}},[[4,5,4]]]);
//# sourceMappingURL=main.bundle.js.map"
`;
